### PR TITLE
Add YOLOv8x Segmentation

### PR DESCRIPTION
**Pull request by zhanghuiyao**
Thank you for your contribution to the MindYOLO repo.
Before submitting this PR, please make sure:

- [x] You have read the [Contributing Guidelines on pull requests](https://github.com/mindspore-lab/mindyolo/blob/master/CONTRIBUTING.md)
- [x] Your code builds clean without any errors or warnings
- [x] You are using approved terminology
- [x] You have added unit tests

## Motivation

1. Add YOLOv8x Seg Model
2. Reconstruct dataset

## Test Plan

(How should this PR be tested? Do you require special setup to run the test or repro the fixed bug?)

## Related Issues and PRs

(Is this PR part of a group of changes? Link the other relevant PRs and Issues here. Use https://help.github.com/en/articles/closing-issues-using-keywords for help on GitHub syntax)

### CODE REVIEW

File: mindyolo/version.py [hunk 1]
```diff
@@ -1,2 +1,2 @@
 """version init"""
-__version__ = "0.1.0"
+__version__ = "0.3.0-dev"
```
Comment (CaitinZhao) [inline mindyolo/version.py:2]: 这个version为什么突然变成0.3了？

File: MODEL_ZOO.md [hunk 2]
```diff
@@ -26,12 +28,16 @@
 | YOLOX  | X                  | D910x8-G | 640       | MS COCO 2017 | 51.6        | 99.1M  | 281.9G | [yaml](https://github.com/mindspore-lab/mindyolo/blob/master/configs/yolox/yolox-x.yaml)         | [weights](https://download.mindspore.cn/toolkits/mindyolo/yolox/yolox-x_300e_map516-52216d90.ckpt)                   |
 | YOLOX  | Darknet53          | D910x8-G | 640       | MS COCO 2017 | 47.7        | 63.7M  | 185.3G | [yaml](https://github.com/mindspore-lab/mindyolo/blob/master/configs/yolox/yolox-darknet53.yaml) | [weights](https://download.mindspore.cn/toolkits/mindyolo/yolox/yolox-darknet53_300e_map477-b5fcaba9.ckpt)                   |
 
-<br>
+## Segmentation
```
Comment (CaitinZhao) [inline MODEL_ZOO.md:31]: quick start里也加下segment的说明

**Comment by CaitinZhao** [comment]
改下代码检查

File: mindyolo/version.py (hunk 1 above)
Comment (zhanghuiyao) [reply mindyolo/version.py:2]: v0.2分支还没改过来 master作为v0.3版本继续开发

**Review by SamitHuang — APPROVED** [review]

File: MODEL_ZOO.md (hunk 2 above)
Comment (zhanghuiyao) [reply MODEL_ZOO.md:31]: seg task兼容原有启动方式，用户通过指定不同.yaml文件来选择不同的task

**Comment by zhanghuiyao** [comment]
> 改下ut适配这次的修改

ok

**Review by SamitHuang — APPROVED** [review]

**Review by CaitinZhao — APPROVED** [review]

**Pull request merged**